spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/NamedConfigMapContextToSourceDataProvider.java

```diff
@@ -72,47 +59,15 @@ public Fabric8ContextToSourceData get() {
 		return context -> {
 
 			NamedConfigMapNormalizedSource source = (NamedConfigMapNormalizedSource) context.normalizedSource();
-			String namespace = context.namespace();
-			String initialConfigMapName = source.name().orElseThrow();
-			String currentConfigMapName = initialConfigMapName;
-			Set<String> propertySourceNames = new LinkedHashSet<>();
-			propertySourceNames.add(initialConfigMapName);
-
-			Map<String, Object> result = new HashMap<>();
-
-			LOG.info("Loading ConfigMap with name '" + initialConfigMapName + "' in namespace '" + namespace + "'");
-			try {
-				Map<String, String> data = getConfigMapData(context.client(), namespace, currentConfigMapName);
-				result.putAll(entriesProcessor.apply(data, context.environment()));
 
-				if (context.environment() != null && source.profileSpecificSources()) {
-					for (String activeProfile : context.environment().getActiveProfiles()) {
-						currentConfigMapName = initialConfigMapName + "-" + activeProfile;
-						Map<String, String> dataWithProfile = getConfigMapData(context.client(), namespace,
-								currentConfigMapName);
-						if (!dataWithProfile.isEmpty()) {
-							propertySourceNames.add(currentConfigMapName);
-							result.putAll(entriesProcessor.apply(dataWithProfile, context.environment()));
-						}
-					}
+			return new NamedSourceData() {
```

Comment:
Since we have all the structure already in place via `NamedSourceData`, all this class has to do is provide its implementation of `dataSupplier`, and it does that via `Fabric8ConfigUtils::configMapsDataByName`.

Since we have support for named secrets also - the same thing happens there. 

And the same thing (via `LabeledNamedSource`) happens for labeled based sources. I will also add, in this PR, support for Labeled Config Maps.